Home screen design refresh

worldeggplant opened the pull request:
It's a bit [WIP], but ready for test.

Left:

- [ ] Clean up
- [ ] Adding icons to Lund as well
- [ ] Add dependencies that require linking to `template/`

worldeggplant committed:
New primary color for hbg

worldeggplant committed:
Added ImageCard and made some slight refactoring. Some cleanup left.

worldeggplant committed:
Added `SELECT_CURRENT_HOME_TAB` to Action typers

Commit:
Merge with develop

Commit:
Update showing selected category/tab in CategoryMapScreen, using new index for tabs

joenas commented on guide-app/src/components/screens/HomeScreen/index.js:
```diff
@@ -89,107 +103,100 @@ class HomeScreen extends Component<Props> {
-  }) => {
-    if (section.hideFooter) {
-      return null;
+    if (this.props.showLoadingSpinner) {
```
Add this to deconstructuring above perhaps?

Commit:
Increased fontsizes for homescreen ImageCard just a bit

Commit:
Only show mapIcon on guides, not guideGroups.

joenas commented on guide-app/src/Nav.js:
```diff
@@ -114,7 +120,7 @@ export default class Nav extends Component<Props> {
         <StatusBar
           translucent={ios}
-          barStyle="light-content"
+          barStyle="dark-content" // @TODO `barStyle` should probably be added to `Colors`
```
On the home screen the bar is purple on Android. Tested adding a `<StatusBar barStyle="dark-content" backgroundColor={Colors.white} />`to `HomeScreen` but then map is white instead.. so we should look at that

worldeggplant replied:
Sounds good, I'll look into it a bit more

joenas commented on guide-app/src/components/shared/ImageCard.js:
```diff
@@ -0,0 +1,132 @@
+      height: 2
+    },
+    shadowOpacity: 0.25,
+    shadowRadius: 3.84,
```
Getting these warnings: 

> View #2697 of type RCTView has a shadow set but cannot calculate shadow efficiently. Consider setting a background color to fix this, or apply the shadow to a more specific component.

Looks like it's this component but not sure what background color to apply

joenas replied:
Added `Colors.white` for now

joenas reviewed (commented):
There seems to be a bug with the bottom bar. When going to the map and then back, it doesn't appear again.

joenas commented on guide-app/src/components/screens/HomeScreen/index.js:
```diff
@@ -89,107 +103,100 @@ class HomeScreen extends Component<Props> {
+    }
+  });
+
+  const items = categories[currentHomeTab].data;
```
This needs a default I believe for `currentHomeTab` when state is new?

worldeggplant replied:
currentHomeTab's default is `0` in the reducer. Do you believe there is a case when the state hasn't been populated, but the HomeScreen is rendered?

joenas replied:
It keeps happening, at least once while switching languages and sometimes at initial startup so yeah somehow.

Commit:
Get rid of warning for FlatList key being number

Commit:
Hide bottom nav on Maps and display when map closes

Commit:
Got rid of warning about missing backgroundColor on ImageCard
> View #393 of type RCTView has a shadow set but cannot calculate shadow efficiently. Consider setting a background color to fix this, or apply the shadow to a more specific component.

Commit:
Removed top padding on HomeScreen pill navigation, to align it with datepicker in Calendar

joenas commented:
> There seems to be a bug with the bottom bar. When going to the map and then back, it doesn't appear again.

Fixed that in 20ad2c286bd9ae3f2915f254397ed2a2fc98ce84

worldeggplant committed:
Safe guard against content not loaded yet

worldeggplant committed:
Fix StatusBar and review fixes

joenas reviewed (approved):
👍

worldeggplant committed:
Remove sharing overlay if sharing fails
